tuf/utils/x509.go

```diff
@@ -252,9 +252,31 @@ func ParsePEMPublicKey(pubKeyBytes []byte) (data.PublicKey, error) {
 			return nil, fmt.Errorf("invalid certificate: %v", err)
 		}
 		return CertToKey(cert), nil
+	case "PUBLIC KEY":
+		keyType, err := keyTypeForPublicKey(pemBlock.Bytes)
```

Comment:
That seems like a good refactor to me; I assume the same for `ParsePEMPrivateKey`?

I was going to bring this up elsewhere but it seems relevant here: I think it's confusing that if you pass in a cert to `ParsePEMPrivateKey` you get back a `data.PublicKey` object containing the cert bytes in the `Public` field instead of the parsed out public key bytes.

I was planning to read through a bit more before suggesting a change, but personally I like the idea of a flow like: certs get loaded/verified -> translated to plain keys -> tuf only deals with plain keys. 

This seems like a very clean separation to me; the only issue would be that you would lose the property that certs get verified right before they're used every time. Perhaps it would make sense, after loading the certs into memory, to keep around a map of `parsedKeys -> originatingCerts` so that at various well-defined points during verification, you can do something like `reverifyCerts(currentlyReleventKeyList)` to find the certs you're using and check them again.

Just a couple of thoughts I've had on making the flow of notary easier to understand. I've just rambled a bit too much for an issue comment, so I'll pull this out into an issue later today...